src/bloomfilter.c

```diff
@@ -18,22 +18,20 @@ BloomFilter *bloomfilter_Create_Malloc(size_t max_num_elem, double error_rate,
         return NULL;
     }
 
-    bf->max_num_elem = max_num_elem;
-    bf->error_rate = error_rate;
-    bf->num_hashes = num_hashes;
-    bf->count_correct = 1;
-    bf->bf_version = BF_CURRENT_VERSION;
-    bf->elem_count = 0;
-    bf->array = NULL;
-    memset(bf->reserved, 0, sizeof(uint32_t) * 32);
-    memset(bf->hash_seeds, 0, sizeof(uint32_t) * 256);
-    memcpy(bf->hash_seeds, hash_seeds, sizeof(uint32_t) * num_hashes);
+    bf->header.max_num_elem = max_num_elem;
+    bf->header.error_rate = error_rate;
+    bf->header.num_hashes = num_hashes;
+    bf->header.bf_version = BF_CURRENT_VERSION;
+    memset(bf->header.hash_seeds, 0, sizeof(uint32_t) * 256);
```

Comment:
Don't use hardcoded constants. Use `sizeof (bf->header.hash_seeds)` if you're clearing a whole member.

Better yet: Don't zero `header.hash_seeds` if you're going to (partially) overwrite it in the next line.

Reply:
I limited myself to rename `bf->hash_seeds` to `bf->header.hash_seeds`, the hardcoded constant was already there. However, I agree, so I have fixed it.

On the other hand, the `memset` *is* necessary, otherwise, we would leave garbage in the `hash_seeds` array that would break the `_assert_comparable` results. And fix that is precisely the objective of this PR.

Reply:
That's not what I meant. Instead of clearing a full array and then copying over it, you can do just what's needed:

```memset (mempcpy (bf->header.hash_seeds, hash_seeds, sizeof (uint32_t) * num_hashes),
                0, sizeof (bf->header.hash_seeds) - sizeof (uint32_t) * num_hashes);
```

But it's a nitpick. OTOH, since many of the field members are zero-initialized, you can get away with using `calloc` to allocate storage for the BF. But again, it's just a nitpick. I pointed out the usage of a hardcoded constant because it's really bad practice in C (this whole library is a mess, anyway).

Reply:
Done.